### PR TITLE
DOC: landing page and minor touches

### PR DESCRIPTION
**Pull request by martinfleis**
An update of a landing page and minor edits of some other pages in the documentation. The link to a rendered version is among Checks below.

### CODE REVIEW

**Review by brendan-ward — COMMENTED** [review]
Thanks for working on this @martinfleis 

A couple minor text suggestions.

File: doc/source/index.md [hunk 1]
```diff
@@ -2,11 +2,47 @@
...
+the `conda-forge` channel but you can also install it from PyPI.
+
+```sh
+conda install dask-geopandas -c conda-forge
+```
+
+```sh
+pip install dask-geopandas
+```
+
```
Comment (jorisvandenbossche) [inline doc/source/index.md:24]: Maybe add a "for more detail .." link to the install page here?

File: doc/source/index.md [hunk 2]
```diff
@@ -2,11 +2,47 @@
...
+As with `dask.dataframe` and `pandas`, the API of `dask_geopandas` mirrors the one of `geopandas`.
+
+```py
+import geopandas
+import dask_geopandas
+
+df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
+dask_df = dask_geopandas.from_geopandas(df, npartitions=4)
+
+ddf.geometry.area.compute()
```
Comment (jorisvandenbossche) [inline doc/source/index.md:36]:
```suggestion
dask_df.geometry.area.compute()
```

(or use `ddf` above)

File: doc/source/installation.md [hunk 3]
```diff
@@ -2,7 +2,29 @@
...
+The best way to install Dask-GeoPandas is using `conda` or `mamba` and `conda-forge` channel:
+
+```sh
+conda install -c conda-forge dask-geopandas
+```
+
+## pip
+
+You can install Dask-GeoPandas with `pip` from PyPI but make sure that your environment contains
+properly installed GeoPandas (note that Dask-GeoPandas does not use `fiona` which therefore doesn't
```
Comment (jorisvandenbossche) [inline doc/source/installation.md:18]: Maybe point to https://geopandas.org/en/stable/getting_started/install.html#installing-with-pip for the geopandas installation (we indeed don't need fiona here in dask-geopandas, but when doing a `pip install geopandas`, we unfortunately still expect fiona to be installed .. so in practice users will run into the same issues I think)

**Review by jorisvandenbossche — COMMENTED** [review]
Nice!

**Comment by martinfleis** [comment]
Thanks both!

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**